test/extended/images/hardprune.go

```diff
@@ -23,323 +23,163 @@ import (
 	imagev1 "github.com/openshift/api/image/v1"
 	"github.com/openshift/library-go/pkg/image/imageutil"
 
+	//imagesutil "github.com/openshift/origin/test/extended/images"
+
 	exutil "github.com/openshift/origin/test/extended/util"
 )
 
 var _ = g.Describe("[sig-imageregistry][Feature:ImagePrune][Serial][Suite:openshift/registry/serial][Local] Image hard prune", func() {
 	defer g.GinkgoRecover()
 	var oc = exutil.NewCLI("prune-images")
-	var originalAcceptSchema2 *bool
-	ctx := context.Background()
```

Comment:
For a first attempt at restoring these tests, I would not worry about hard pruning. Eventually we'll want this test back, but it isn't as pressing as ensuring the core pruning functions work.

@dmage would you agree?

Reply:
Yes, I agree.

I'm fine to slightly update this test even if it stays Skipped if that helps to remove legacy code. We'll never have the docker socket anyway.

Reply:
OK, I will not spend too much effort on hard pruning now. Should I revert my changes?

Reply:
Let's focus on `oc adm prune images` and remove hard pruner changes from this PR. Keep them for a follow-up PR.

btw, you shouldn't have back merges in your PR, so you'll need to squash commits.

Reply:
OK, I have restored most part of hard prune, just leave some basic functions change which maybe used by prune.go.